feat(http): Add timeout option to HTTP requests (#57194)
angular/angular: adev/src/content/guide/http/making-requests.md

## Handling request failure

There are three ways an HTTP request can fail:

* A network or connection error can prevent the request from reaching the backend server.
* A request didn't respond in time when the timeout option was set.
* The backend can receive the request but fail to process it, and return an error response.

> `HttpClient` captures both kinds of errors in an `HttpErrorResponse` which it returns through the `Observable`'s error channel. Network and timeout errors have a `status` code of `0` and an `error` which is an instance of [`ProgressEvent`](https://developer.mozilla.org/docs/Web/API/ProgressEvent). Backend errors have the failing `status` code returned by the backend, and the error response as the `error`. Inspect the response to identify the error's cause and the appropriate action to handle the error.

Reviewer: minor: "both" should be updated as there are more than 2.

```suggestion
`HttpClient` captures all of the above errors in an `HttpErrorResponse` which it returns through the `Observable`'s error channel. Network and timeout errors have a `status` code of `0` and an `error` which is an instance of [`ProgressEvent`](https://developer.mozilla.org/docs/Web/API/ProgressEvent). Backend errors have the failing `status` code returned by the backend, and the error response as the `error`. Inspect the response to identify the error's cause and the appropriate action to handle the error.
```
